graphdatascience/tests/unit/test_aura_api.py

```diff
@@ -0,0 +1,186 @@
+
+
+def test_derive_tenant(requests_mock: Mocker) -> None:
+    mock_auth_token(requests_mock)
+
+    requests_mock.get(
+        "https://api.neo4j.io/v1/tenants",
+        json={"data": [{"id": "6981ace7-efe8-4f5c-b7c5-267b5162ce91", "name": "Production"}]},
+    )
+
+    AuraApi(client_id="", client_secret="")
+
+
+def test_raise_on_missing_tenant(requests_mock: Mocker) -> None:
+    mock_auth_token(requests_mock)
+
+    requests_mock.get(
+        "https://api.neo4j.io/v1/tenants",
+        json={
+            "data": [
+                {"id": "6981ace7-efe8-4f5c-b7c5-267b5162ce91", "name": "Production"},
+                {"id": "YOUR_TENANT_ID", "name": "Staging"},
+                {"id": "da045ab3-3b89-4f45-8b96-528f2e47cd13", "name": "Development"},
+            ]
+        },
+    )
+
+    with pytest.raises(RuntimeError, match="This account has access to multiple tenants"):
+        AuraApi(client_id="", client_secret="")
+
+
+def test_list_instance(requests_mock: Mocker) -> None:
+    api = AuraApi(client_id="", client_secret="", tenant_id="YOUR_TENANT_ID")
+
+    mock_auth_token(requests_mock)
+    requests_mock.get(
+        "https://api.neo4j.io/v1/instances/id0",
+        json={
+            "data": {
+                "id": "2f49c2b3",
+                "name": "Production",
+                "status": "running",
+                "tenant_id": "YOUR_TENANT_ID",
+                "cloud_provider": "gcp",
+                "connection_url": "YOUR_CONNECTION_URL",
+                "region": "europe-west1",
+                "type": "enterprise-db",
+                "memory": "8GB",
+            }
+        },
+    )
+
+    result = api.list_instance("id0")
+
+    assert result and result.id == "2f49c2b3"
+
+
+def test_list_instance_missing_memory_field(requests_mock: Mocker) -> None:
+    api = AuraApi(client_id="", client_secret="", tenant_id="some-tenant")
+
+    mock_auth_token(requests_mock)
+    requests_mock.get(
+        "https://api.neo4j.io/v1/instances/id0",
+        json={
+            "data": {
+                "cloud_provider": "gcp",
+                "connection_url": None,
+                "id": "a10fb995",
+                "name": "gds-session-foo-bar",
+                "region": "europe-west1",
+                "status": "creating",
+                "tenant_id": "046046d1-6996-53e4-8880-5b822766e1f9",
+                "type": "enterprise-ds",
+            }
+        },
+    )
+
+    result = api.list_instance("id0")
+
+    assert result and result.id == "a10fb995"
+    assert result.memory == ""
+
+
+def test_list_missing_instance(requests_mock: Mocker) -> None:
+    api = AuraApi("", "", tenant_id="some-tenant")
+
+    mock_auth_token(requests_mock)
+
+    requests_mock.get(
+        "https://api.neo4j.io/v1/instances/id0",
+        status_code=404,
+        reason="Not Found",
+        json={"errors": [{"message": "DB not found: id0", "reason": "db-not-found"}]},
+    )
+
+    assert api.list_instance("id0") is None
```

Comment:
see above, an indication for the user might be useful

Reply:
This is not a user-facing method. We use it from two places currently:

- `connect`, which does raise an error when the to-be-connected session is not found
- `wait_for_instance_running` which is called from `create_gds` and there we just ignore the return value. That's a bug that I will fix on this PR.

I suggest making the behaviour such that if we get `None` back then we will raise an error indicating that there was a failure in creating the session.

I don't think we need to change how `list_instance()` works though.